source/fundamentals/crud/query-document.txt

```diff
@@ -0,0 +1,331 @@
+   :depth: 1
+   :class: singlecol
+
+.. _query_document_golang:
+
+Overview
+--------
+
+In this guide, you can learn how to specify a query to match a subset
+of documents.
+
+To match a subset of documents, specify a **query filter** containing
+your **match criteria**. Match criteria consists of the fields and
+values you want matched. A query filter contains at least one set
+of match criteria to determine which documents to include in the
+resulting set.
+
+In a query filter, you can match fields with :ref:`literal values
+<go-literal-values>` or with **query operators**. Query
+operators allow you to perform mathematical or logical operations to
+locate documents within a collection.
+
+The following shows the match criteria format with literal values:
+
+.. code-block:: go
+   :copyable: false
+
+   filter := bson.D{{"<field>", "<value>"}}
+
+The following shows the match criteria format with a query operator:
+
+.. code-block:: go
+   :copyable: false
+   
+   filter := bson.D{{"<field>", bson.D{{"<operator>", "<value>"}}}}
+
+This guide explains how to use :ref:`literal values <go-literal-values>`
+and :ref:`query operators <query-operators-go>` with the ``Find()``
+function.
+
+Sample Data
+~~~~~~~~~~~
+
+To run the examples in this guide, load these documents into the
+``ratings`` collection of the ``tea`` database with the following
+snippet:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/query.go
+   :language: go
+   :dedent:
+   :start-after: begin insert docs
+   :end-before: end insert docs
+
+.. include:: /includes/fundamentals/automatic-db-coll-creation.rst
+   
+Each document contains a rating for a type of tea and the vendors that
+carry them, which corresponds to the ``rating``, ``type``, and
+``vendor`` fields.
+
+.. include:: /includes/fundamentals/truncated-id.rst
+
+.. _go-literal-values:
+
+Literal Values
+--------------
+
+Literal value query filters return documents with an exact match to your
+match criteria.
+
+.. tip::
+
+   If you specify an empty query filter, CRUD operations match all the
+   documents in a collection.
+
+Example
+~~~~~~~
+
+The following example matches documents where the ``type`` is "Oolong":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/query.go
+   :language: go
+   :dedent:
+   :start-after: begin literal
+   :end-before: end literal
+
+After running this example, the output resembles the following:
+
+.. code-block:: none
+   :copyable: false
+
+   [{_id ObjectID("...")} {type Oolong} {rating 7} {vendor [C]}]
+
+.. tip::
+
+   Literal value queries returns the same value as the ``$eq``
+   comparison operator. The following query filters produce the same
+   result:
+
+   .. code-block:: go
+   
+      filter := bson.D{{"type", "Oolong"}}
+
+   .. code-block:: go
+
+      filter := bson.D{{"type", bson.D{{"$eq", "Oolong"}}}}
+
+.. _query-operators-go:
+
+Comparison
+----------
+
+Comparison operators compare the value in a document to the specified
```

Comment:
Can we use a different word here than "compare"? Again, we want to avoid a circular definition.